include/openthread/ip6.h

```diff
@@ -213,14 +213,17 @@ typedef struct otSockAddr
  */
 typedef struct otMessageInfo
 {
-    otIp6Address mSockAddr;      ///< The local IPv6 address.
-    otIp6Address mPeerAddr;      ///< The peer IPv6 address.
-    uint16_t     mSockPort;      ///< The local transport-layer port.
-    uint16_t     mPeerPort;      ///< The peer transport-layer port.
-    const void * mLinkInfo;      ///< A pointer to link-specific information.
-    uint8_t      mHopLimit;      ///< The IPv6 Hop Limit value. Only applies if `mAllowZeroHopLimit` is FALSE.
-                                 ///< If `0`, IPv6 Hop Limit is default value `OPENTHREAD_CONFIG_IP6_HOP_LIMIT_DEFAULT`.
-                                 ///< Otherwise, specifies the IPv6 Hop Limit.
+    otIp6Address mSockAddr; ///< The local IPv6 address.
+    otIp6Address mPeerAddr; ///< The peer IPv6 address.
+    uint16_t     mSockPort; ///< The local transport-layer port.
+    uint16_t     mPeerPort; ///< The peer transport-layer port.
+    const void * mLinkInfo; ///< A pointer to link-specific information.
+    uint32_t
+        mVersionClassFlow; ///< Version, class, and flow information. If version is not 6 (0b110), a default is used.
+                           ///< DSCP bits of this field are overridden by the priority attached to the otMessage.
```

Comment:
Sorry for missing this comment earlier. The purpose of this is to allow one to process IP packets in an ECN-aware way --- to mark packets as ECN-capable when sending them out, and checking if they were marked as "congestion experienced" when receiving them.

The `MessageInfo` class is tied to `otMessageInfo`, so adding the field required changing it in the public API.

This is primarily to be able to read and write the traffic class --- specifically the last two bits of the traffic class, corresponding to ECN. I bundled it with the version and class since that's how the IPv6 implementation dealt with it. Should I just keep it to specifying ECN for now? Or should I split the DCSP, ECN, and Flow bits into separate fields?

Reply:
The latest update changes this to just have a field `mEcn` to avoid redundantly specifying DCSP and version. I figured that if we ever need to specify the flow, we can add `mFlow` at that time. I hope this addresses your concern that bundling these together into a single field might be confusing for the programmer.